lib/premailer/adapter/nokogiri.rb

```diff
@@ -150,9 +150,9 @@ def write_unmergable_css_rules(doc, unmergable_rules) # :nodoc:
           style_tag = "<style type=\"text/css\">\n#{styles}</style>"
           unless (body = doc.search('body')).empty?
             if doc.at_css('body').children && !doc.at_css('body').children.empty?
-              doc.at_css('body').children.before(::Nokogiri::XML.fragment(style_tag))
+              doc.at_css('head').children.after(::Nokogiri::XML.fragment(style_tag))
             else
-              doc.at_css('body').add_child(::Nokogiri::XML.fragment(style_tag))
+              doc.at_css('body').add_next_sibling(::Nokogiri::XML.fragment(style_tag))
```

Comment:
This will insert the style node after body (outside of it). I don't think this would be recognized any more than having it inside the body: since `<head><style>` is recognized everywhere now, is it even needed?